### PR TITLE
Fix shellcheck failures on ginkgo-e2e.sh

### PR DESCRIPTION
**Pull request by xichengliudui**
/kind cleanup
/priority backlog
/kind cleanup
/sig testing

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/ginkgo-e2e.sh [hunk 1]
```diff
@@ -48,11 +48,13 @@ source "${KUBE_ROOT}/cluster/kube-util.sh"
 function detect-master-from-kubeconfig() {
     export KUBECONFIG=${KUBECONFIG:-$DEFAULT_KUBECONFIG}
 
-    local cc=$("${KUBE_ROOT}/cluster/kubectl.sh" config view -o jsonpath="{.current-context}")
-    if [[ ! -z "${KUBE_CONTEXT:-}" ]]; then
+    local cc
```
Comment (xichengliudui) [inline hack/ginkgo-e2e.sh]: Ref: `https://github.com/koalaman/shellcheck/wiki/SC2155` [1]

File: hack/ginkgo-e2e.sh [hunk 2]
```diff
@@ -48,11 +48,13 @@ source "${KUBE_ROOT}/cluster/kube-util.sh"
...
+    cc=$("${KUBE_ROOT}/cluster/kubectl.sh" config view -o jsonpath="{.current-context}")
+    if [[ -n "${KUBE_CONTEXT:-}" ]]; then
       cc="${KUBE_CONTEXT}"
     fi
-    local cluster=$("${KUBE_ROOT}/cluster/kubectl.sh" config view -o jsonpath="{.contexts[?(@.name == \"${cc}\")].context.cluster}")
+    local cluster
```
Comment (xichengliudui) [inline hack/ginkgo-e2e.sh]: [1]

File: hack/ginkgo-e2e.sh [hunk 3]
```diff
@@ -138,8 +140,9 @@ fi
 # The --host setting is used only when providing --auth_config
 # If --kubeconfig is used, the host to use is retrieved from the .kubeconfig
 # file and the one provided with --host is ignored.
-# Add path for things like running kubectl binary.
-export PATH=$(dirname "${e2e_test}"):"${PATH}"
+# Add path for things like running kubectl binary. 
+PATH=$(dirname "${e2e_test}"):"${PATH}"
+export PATH
```
Comment (xichengliudui) [inline hack/ginkgo-e2e.sh]: [1]

File: hack/ginkgo-e2e.sh [hunk 4]
```diff
@@ -86,8 +88,8 @@ fi
 if [[ "${KUBERNETES_PROVIDER}" == "gce" ]]; then
   set_num_migs
   NODE_INSTANCE_GROUP=""
-  for ((i=1; i<=${NUM_MIGS}; i++)); do
-    if [[ ${i} == ${NUM_MIGS} ]]; then
+  for ((i=1; i<=NUM_MIGS; i++)); do
```
Comment (xichengliudui) [inline hack/ginkgo-e2e.sh]: Ref: `https://github.com/koalaman/shellcheck/wiki/SC2004`

File: hack/ginkgo-e2e.sh [hunk 5]
```diff
@@ -86,8 +88,8 @@ fi
...
+    if [[ ${i} == "${NUM_MIGS}" ]]; then
```
Comment (xichengliudui) [inline hack/ginkgo-e2e.sh]: Ref: `https://github.com/koalaman/shellcheck/wiki/SC2053`

**Review by fejta — COMMENTED** [review]
Please remove the shellcheck exception if you're going to update this file

https://github.com/kubernetes/kubernetes/blob/36ccff1b27b6aee5c5599334842cb2b8cf8063f3/hack/.shellcheck_failures#L24

**Comment by xichengliudui** [comment]
@fejta  Thank you. I forgot to delete it

**Review by fejta — COMMENTED** [review]
/lgtm
/approve
/retest

**Comment by fejta** [comment]
> /test pull-kubernetes-e2e-gce-100-performance

A `/retest` command is an easier way to accomplish the same thing

**Comment by xichengliudui** [comment]
Thanks for the warning.

**Comment by xichengliudui** [comment]
Resolved file conflicts.

**Review by fejta — COMMENTED** [review]
/lgtm

**Pull request merged**